src/main/java/seedu/address/model/student/UniqueStudentList.java

```diff
@@ -241,52 +493,163 @@ create a class from the start time and check to see if it exceeds the startTime
-                    classToCompare.endTime.plusMinutes(tr.duration));
+        LocalTime startTimeFromTr = tr.startTimeRange;
+        LocalTime endTimeFromTr = startTimeFromTr.plusMinutes(tr.duration);
+        LocalTime startTimeFromClass = classToCompare.endTime;
+        LocalTime endTimeFromClass = startTimeFromClass.plusMinutes(tr.duration);
+        LocalTime startTimeFromCurrTime = currTime;
+        LocalTime endTimeFromCurrTime = startTimeFromCurrTime.plusMinutes(tr.duration);
+
+        if (!classToCompare.date.equals(currDate)) {
+            newClass = new Class(currDate, startTimeFromTr, endTimeFromTr);
+        } else if (currTime.compareTo(tr.endTimeRange) >= 0) {
+            // if the currentTime is after and including the end of the time range, just go to the next day
+            newClass = new Class(currDate.plusDays(1), startTimeFromTr, endTimeFromTr);
+        } else if (currTime.compareTo(tr.startTimeRange) <= 0) {
+            // if the currentTime is before and including the start time range
+            if (classToCompare.endTime.compareTo(tr.startTimeRange) <= 0) {
+                // the sitaution where the endTime of the class is before the tr.startTimeRange
+                newClass = new Class(currDate, startTimeFromTr, endTimeFromTr);
+            } else if (classToCompare.endTime.compareTo(tr.startTimeRange) > 0
+                    && classToCompare.endTime.compareTo(tr.endTimeRange) < 0
+                    && classToCompare.startTime.compareTo(tr.startTimeRange) <= 0) {
+                // getting the case where the endTime is after the tr.startTimeRange but satisfies the above conditions
+                if (endTimeFromClass.compareTo(tr.endTimeRange) <= 0) {
+                    newClass = new Class(currDate, startTimeFromClass, endTimeFromClass);
+                }
+            } else if (classToCompare.endTime.compareTo(tr.startTimeRange) > 0
+                    && classToCompare.endTime.compareTo(tr.endTimeRange) < 0
+                    && classToCompare.startTime.compareTo(tr.startTimeRange) > 0) {
+                if (endTimeFromTr.compareTo(classToCompare.startTime) <= 0) {
+                    // trying to fit a slot from the start of the class
+                    newClass = new Class(currDate, startTimeFromTr, endTimeFromTr);
+                } else if (classToCompare.endTime.plusMinutes(tr.duration).compareTo(tr.endTimeRange) <= 0) {
+                    // fit a class after the end of the class
+                    newClass = new Class(currDate, startTimeFromClass, endTimeFromClass);
+                }
+            } else if (classToCompare.startTime.compareTo(tr.startTimeRange) > 0) {
+                if (endTimeFromTr.compareTo(classToCompare.startTime) <= 0) {
+                    newClass = new Class(currDate, startTimeFromTr, endTimeFromTr);
+                }
+            } else if (classToCompare.startTime.compareTo(tr.endTimeRange) >= 0) {
```

Comment:
this if block will never be gone into as the check in line 583 has covered the situation here.

Reply:
Resolved!